Topic 0/src/abstractFactory/DatabaseFactory.java

```diff
@@ -0,0 +1,6 @@
+package abstractFactory;
+
+public abstract class DatabaseFactory {
+	abstract MySql getMySqlDb(String mySql);
+	abstract Oracle getOracleDb(String oracle);
```

Comment:
Another improvement would be use only one method returning an interface, with two implementations MySQL and Oracle.